src/page/modal-demo.js

```diff
@@ -22,6 +24,12 @@ export class ModalDemo extends React.Component {
     this.setState({ [modalName]: true });
   }
 
+  onSwitchChange(index) {
+    let switches = this.state.switches;
```

Comment:
We should use `_.assign` here so we don't ever overwrite this.state.variableName directly